scim/bulkRecords.bal

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2022, WSO2 LLC. (http://www.wso2.com).
+
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+# + failOnErrors - No. of errors to fail the bulk request
+# + schemas - URIs of the SCIM schemas used in the request  
+# + Operations - List of operations
+public type Bulk record {
+    int failOnErrors?;
+    string[] schemas;
+    Operation[] Operations;
+};
+
+# + method - method of the operation  
+# + path - path of the operation 
+# + bulkId - bulkId of the operation 
+# + data - data of the operation
+public type Operation record {
+    string method;
+    string path;
+    string bulkId?;
+    SCIMUser|BulkUserUpdate|BulkUserReplace|SCIMGroup|BulkGroupUpdate|BulkGroupReplace data?;
+};
```

Comment:
version is an optional parameter of operation in the  bulk request
https://www.rfc-editor.org/rfc/rfc7644.html#page-16:~:text=version%20%20The%20current%20resource%20version.%20%20Version%20MAY%20be%20used%20if%20the%0A%20%20%20%20%20%20%20%20%20service%20provider%20supports%20entity%2Dtags%20(ETags)%20(Section%C2%A02.3%20of%0A%20%20%20%20%20%20%20%20%20%5BRFC7232%5D)%20and%20%22method%22%20is%20%22PUT%22%2C%20%22PATCH%22%2C%20or%20%22DELETE%22.

Reply:
ack